lib/new_relic/agent/instrumentation/active_record_subscriber.rb

```diff
@@ -91,21 +91,26 @@ def active_record_config(payload)
           end
 
           return unless connection_id = payload[:connection_id]
-          connection = nil
 
           ::ActiveRecord::Base.connection_handler.connection_pool_list.each do |handler|
-            # RE: https://github.com/newrelic/newrelic-ruby-agent/issues/507
-            spec_config = handler.respond_to?(:spec) && handler.spec && handler.spec.config
-            return spec_config if using_makara?(spec_config)
-
-            connection = handler.connections.detect do |conn|
-              conn.object_id == connection_id
-            end
-
-            break if connection
```

Comment:
NOTE that this loop through the connection pool list is only reached for Rails 5 and below.

These changes do the following:

- Instead of returning directly from the loop OR breaking and then returning, simply return from the loop in both early exit cases. This saves us from needing to have `connection` defined outside of the loop as well.
- Check for `#connections` - as this code always has - before `#spec`.
- Perform a `use_spec_config?` method to encompass ALL cases that leverage the handler spec config. For now, only Makara in supported in this way, but there could be others added in future.